src/test/kotlin/dev/stalla/model/podcastindex/GeographicLocationTest.kt

```diff
@@ -0,0 +1,284 @@
+        assertThat(GeographicLocation.of("geo:12.34,56.78;u=invalid")).isNotNull().all {
+            prop(GeographicLocation::latitude).isEqualTo(12.34)
+            prop(GeographicLocation::longitude).isEqualTo(56.78)
+            prop(GeographicLocation::altitude).isNull()
+            prop(GeographicLocation::crs).isNull()
+            prop(GeographicLocation::uncertainty).isNull()
+            prop(GeographicLocation::parameters).hasSize(1)
+            hasParameterWithValue("u", "invalid")
+        }
+    }
+
+    @Test
+    fun `should not parse a Geo URI with only a latitude`() {
+        assertThat(GeographicLocation.of("geo:12.34")).isNull()
+    }
+
+    @Test
+    fun `should not parse a Geo URI when there is only the scheme`() {
+        assertThat(GeographicLocation.of("geo:")).isNull()
+    }
+
+    @Test
+    fun `should not parse an URI that is not a Geo URI`() {
+        assertThat(GeographicLocation.of("https://stalla.dev")).isNull()
+    }
+
+    @Test
+    fun `should not parse a Geo URI when the scheme is missing`() {
+        assertThat(GeographicLocation.of("48.198634,16.371648;crs=wgs84;u=40")).isNull()
+    }
+
+    @Test
+    fun `should parse a Geo URI and decode special chars in param value`() {
+        assertThat(GeographicLocation.of("geo:12.34,56.78;param=with%20%3d%20special%20&%20chars")).isNotNull().all {
+            prop(GeographicLocation::latitude).isEqualTo(12.34)
+            prop(GeographicLocation::longitude).isEqualTo(56.78)
+            prop(GeographicLocation::altitude).isNull()
+            prop(GeographicLocation::crs).isNull()
+            prop(GeographicLocation::uncertainty).isNull()
+            prop(GeographicLocation::parameters).hasSize(1)
+            hasParameterWithValue("param", "with = special & chars")
+        }
+    }
+
+    @Test
+    fun `should parse a Geo URI and match multiple parameters correctly`() {
+        assertThat(GeographicLocation.of("geo:12.34,45.67,-21.43;crs=theCrs;u=12.0;param=value;param2=value2")).isNotNull().all {
+            prop(GeographicLocation::latitude).isEqualTo(12.34)
+            prop(GeographicLocation::longitude).isEqualTo(45.67)
+            prop(GeographicLocation::altitude).isEqualTo(-21.43)
+            prop(GeographicLocation::crs).isEqualTo("theCrs")
+            prop(GeographicLocation::uncertainty).isEqualTo(12.0)
+            prop(GeographicLocation::parameters).hasSize(2)
+            hasParameterWithValue("param", "value")
+            hasParameterWithValue("param2", "value2")
+        }
+    }
+
+    @Test
+    fun `should match Geo URIs when WGS-84 special pole case applies correctly`() {
+        val geoLocation1 = GeographicLocation.of("geo:90,-22.43;crs=WGS84")
+        val geoLocation2 = GeographicLocation.of("geo:90,46;crs=WGS84")
+        assertAll {
+            assertThat(geoLocation1).isNotNull()
+            assertThat(geoLocation2).isNotNull()
+            assertThat(geoLocation1!!.match(geoLocation2!!)).isTrue()
```

Comment:
We could avoid the `!!`s if we had `val geoLocation1 = checkNotNull(GeographicLocation.of("geo:90,-22.43;crs=WGS84"))` etc.  We lose some nicety but we do have tests to check parsing valid stuff doesn't return null already, so I think it's a valid compromise